classes/classes/Saves.as

```diff
@@ -1366,6 +1358,98 @@ public function onDataLoaded(evt:Event):void
 	//playerMenu();
 }
 
+/**
+ * Upgrade loaded saveFile.data object to most recent version
+ * so the loadGameObject can proceed without hacks
+ */
+private function unFuckSaveDataBeforeLoading(data:Object):void {
+	if (typeof data.lowerBodyPart != 'object') {
+		if (data.legCount == undefined) {
+			switch (data.lowerBody) {
+				case LOWER_BODY_TYPE_DRIDER_LOWER_BODY:
+					data.legCount = 8;
+					break;
+				case LOWER_BODY_TYPE_CENTAUR:
+					data.legCount  = 4;
+					data.lowerBody = LOWER_BODY_TYPE_HOOFED;
+					break;
+				case LOWER_BODY_TYPE_PONY:
+					data.legCount = 4;
+					break;
+				case LOWER_BODY_TYPE_DEERTAUR:
+					data.legCount  = 4;
+					data.lowerBody = LOWER_BODY_TYPE_CLOVEN_HOOFED;
+					break;
+				case LOWER_BODY_TYPE_NAGA:
+					data.legCount = 1;
+					break;
+				case LOWER_BODY_TYPE_GOO:
+					data.legCount = 1;
+					break;
+				default:
+					data.legCount = 2;
+			}
+		}
+		data.lowerBodyPart = {
+			type    : data.lowerBody,
+			legCount: data.legCount
+		};
+	}
+	if (typeof data.skin != 'object') {
+		//Convert from old skinDesc to new skinAdj + skinDesc!
+		var skinDesc:* = data.skinDesc;
+		var skinAdj:*  = data.skinAdj;
+		var skinType:* = data.skinType;
+		for each (var legacyAdj:String in ["smooth", "thick", "rubber", "latex", "slimey"]) {
+			if (skinDesc.indexOf(legacyAdj) != -1) {
+				skinAdj = legacyAdj;
+				if (skinType == SKIN_TYPE_FUR) {
+					skinDesc = "fur";
+				} else if (skinType == SKIN_TYPE_GOO) {
+					skinDesc = "goo";
+				} else if ([
+							   SKIN_TYPE_DRAGON_SCALES,
+							   SKIN_TYPE_FISH_SCALES,
+							   SKIN_TYPE_LIZARD_SCALES
+						   ].indexOf(skinType) >= 0) {
+					skinDesc = "scales";
+				} else {
+					skinDesc = "skin";
+				}
+			}
+		}
+		data.skin = {
+			type    : data.skinType || SKIN_TYPE_PLAIN,
+			desc    : skinDesc || "skin",
+			adj     : skinAdj || "",
+			color   : data.skinTone || "albino",
+			furColor: data.furColor || "no"
+		}
+	}
+	if (typeof data.clawPart !== "object") {
```

Comment:
I try not to rename game-level or player-level members because they can be resolved in `[parser]` tags. At least not in this commit.